dynawo/sources/Modeler/DataInterface/DYNCriteria.cpp

```diff
@@ -203,52 +202,46 @@ LoadCriteria::checkCriteria(double t, bool finalStep, const boost::shared_ptr<ti
+    double p = load->getP();
+    if (!params_->getVoltageLevels().empty()) {
+      if (alreadyChecked.find(load->getID()) != alreadyChecked.end()) continue;
+      for (std::vector<criteria::CriteriaParamsVoltageLevel>::const_iterator itVl = params_->getVoltageLevels().begin(),
+          itVlEnd = params_->getVoltageLevels().end();
+          itVl != itVlEnd; ++itVl) {
+        const criteria::CriteriaParamsVoltageLevel& vl = *itVl;
+        if ((vl.hasUMaxPu() || vl.hasUMinPu()) && load->getBusInterface()) {
+          double v = load->getBusInterface()->getStateVarV();
+          double vNom = load->getBusInterface()->getVNom();
+          if (vl.hasUMaxPu() && v > vl.getUMaxPu()*vNom) continue;
+          if (vl.hasUMinPu() && v < vl.getUMinPu()*vNom) continue;
         }
-        if (params_->hasPMin() && p < params_->getPMin()) {
-          std::shared_ptr<FailingCriteria> loadFailingCriteria(new LoadFailingCriteria(Bound::MIN,
-                                                                (*it)->getID(),
-                                                                p,
-                                                                params_->getPMin(),
-                                                                params_->getId()));
-          distanceToLoadFailingCriteriaMap.insert({loadFailingCriteria->getDistance(), loadFailingCriteria});
-          Message mess = DYNLog(SourceUnderPower, (*it)->getID(), p, params_->getPMin(), params_->getId());
-          failingCriteria_.push_back(std::make_pair(t, mess.str()));
-          isCriteriaOk &= false;
-        }
-      } else {
-        loadToSourcesAddedIntoSumMap.insert({p, *it});
-        if (alreadySummed.find((*it)->getID()) != alreadySummed.end()) continue;
-        alreadySummed.insert((*it)->getID());
-        sum+=p;
-        atLeastOneEligibleLoadWasFound = true;
+        checkCriteriaInLocalValueOrSumType(t,
+                                            load,
+                                            p,
+                                            loadToSourcesAddedIntoSumMap,
+                                            distanceToLoadFailingCriteriaMap,
+                                            alreadyChecked,
+                                            isCriteriaOk,
+                                            sum,
+                                            atLeastOneEligibleLoadWasFound);
+        break;
       }
+    } else {
+      checkCriteriaInLocalValueOrSumType(t,
+                                          load,
+                                          p,
+                                          loadToSourcesAddedIntoSumMap,
+                                          distanceToLoadFailingCriteriaMap,
+                                          alreadyChecked,
+                                          isCriteriaOk,
+                                          sum,
+                                          atLeastOneEligibleLoadWasFound);
```

Comment:
Shouldn't we add a break here too?

Reply:
Unlike the checkCriteriaInLocalValueOrSumType call just above, this call is outside the for loop we don't need a break here.

Reply:
How come that we don't test anymore the voltage levels for the sum case?

Reply:
ok got it